File: src/009/main.c
# include <stdio.h>

typedef unsigned int u32;

struct triplet {
	u32 a;
	u32 b;
	u32 c;
} get_pythagorean_triplet (u32 sum) {
	for (u32 a = 1; a < sum / 3; a++) {
		for (u32 b = a + 1; b < sum / 2; b++) {
			u32 c = sum - a - b;
			if (a * a + b * b == c * c) {
				return (struct triplet) {a, b, c};
			}
		}
	}
	return (struct triplet) {0, 0, 0};
}

int main () {
	const u32 SUM = 1000;
	const struct triplet TRIPLET = get_pythagorean_triplet (SUM);
	printf ("%u\n", TRIPLET.a * TRIPLET.b * TRIPLET.c);
}

File: src/009/main.rs

struct Triplet {
	a: u32,
	b: u32,
	c: u32
}

fn get_pythagorean_triplet (sum: u32) -> Triplet {
	for a in 1..(sum / 3) {
		for b in (a + 1)..(sum / 2) {
			let c = sum - a - b;
			if a * a + b * b == c * c {
				return Triplet {a: a, b: b, c: c};
			}
		}
	}
	return Triplet {a: 0, b: 0, c: 0};
}

fn main () {
	const SUM: u32 = 1000;
	let triplet: Triplet = get_pythagorean_triplet (SUM);
	println! ("{}", triplet.a * triplet.b * triplet.c);
}

